code/src/java/pcgen/gui2/facade/EquipNode.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Extracted from code that was
+ * Copyright 2008 Connor Petty <cpmeister@users.sourceforge.net>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+package pcgen.gui2.facade;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+
+import org.apache.commons.lang3.StringUtils;
+
+import pcgen.cdom.enumeration.StringKey;
+import pcgen.core.BodyStructure;
+import pcgen.core.Equipment;
+import pcgen.core.character.EquipSlot;
+import pcgen.facade.core.EquipmentFacade;
+
+/**
+ * The Class {@code EquipNodeImpl} represents a node in the equipping
+ * tree. It may be an item of equipment or a slot that may be filled.
+ * EquipNodeImpl objects are immutable.
+ */
+public class EquipNode implements Comparable<EquipNode>
+{
+	public static enum NodeType
+	{
+		/** This is a body location which can contain other slots rather 
+		 * than equipment. */
+		BODY_SLOT,
+	
+		/** This is a node which shows an empty slot that takes a specific 
+		 * equipment type (e.g. Ring). It is removed from the tree when the 
+		 * slot it represents is full and added back in when there is space 
+		 * in the slot. */
+		PHANTOM_SLOT,
+	
+		/** This is a node representing an equipped item of equipment. */
+		EQUIPMENT;
+	}
+
+	private static final NumberFormat FMT = new DecimalFormat("00");
+	private final NodeType nodeType;
+	private final BodyStructure bodyStructure;
+	private Equipment equipment;
+	EquipNode parent;
```

Comment:
why are some things not private here?

Reply:
This is a minimal change and that's how they were before